keras/src/export/export_utils.py

```diff
@@ -17,19 +25,24 @@ def get_input_signature(model):
             "The model provided has not yet been built. It must be built "
             "before export."
         )
+        
     if isinstance(model, models.Functional):
-        input_signature = [
-            tree.map_structure(make_input_spec, model._inputs_struct)
-        ]
+        input_signature = tree.map_structure(make_input_spec, model._inputs_struct)
     elif isinstance(model, models.Sequential):
         input_signature = tree.map_structure(make_input_spec, model.inputs)
     else:
+        # For subclassed models, try multiple approaches
         input_signature = _infer_input_signature_from_model(model)
-        if not input_signature or not model._called:
-            raise ValueError(
-                "The model provided has never called. "
-                "It must be called at least once before export."
-            )
+        if not input_signature:
+            # Fallback: Try to get from model.inputs if available
+            if hasattr(model, 'inputs') and model.inputs:
```

Comment:
can you explain what case you are trying to cover here? when will a model have `inputs` but _infer_input_signature_from_model won't work?